### DISCUSSION TITLE
Logging with Microsoft.Extensions.Logging.Abstractions?

### DISCUSSION THREAD

**Discussion by Zerowalker** [quartznet/quartznet#2064] [Q&A] [answered]
I don't quite understand the logging, reading the guide links to an old logging api that's deprecated by `Microsoft.Extensions.Logging.Abstractions`.
https://github.com/damianh/LibLog

So what's the recommended way to do this?

My aim is to have a logger that writes to a database.
Currently i use a workaround by creating a `ILogger` with a database connection,
then give the job the logger and grab it when it's run with `context.JobDetail.JobDataMap.Get("logger");`

But that only works for my manual logging and not the Quarts own, so I don't think this is a good way to do this.

**Comment by lahma** [comment, marked as answer]
If you check [the ASP.NET Core example](https://github.com/quartznet/quartznet/tree/3.x/src/Quartz.Examples.AspNetCore), you can see how MS DI is configured and it's using MS Logging too. [The documentation site](https://www.quartz-scheduler.net/documentation/quartz-3.x/packages/microsoft-di-integration.html) also has more elaborate examples.

**Reply by Zerowalker** [reply]
Don't really understand the logging setup there.
Is this the part that adds the logger: https://github.com/quartznet/quartznet/blob/3.x/src/Quartz.Examples.AspNetCore/Startup.cs#L45
Where does `UseSerilog` come from?

I'm using `.NET Core 7.0` and not `ASP.NET Core`, not familiar with the ASP parts.
I have setup a `HostingService` before though so i got it partly working, but i can't make the logger work at all,
though it does print the basic stuff like:

```
 Microsoft.Hosting.Lifetime[0]
 Application started. Press Ctrl+C to shut down.
```

thanks:)

**Reply by lahma** [reply]
Maybe give a link to public repo that shows the problem.

**Reply by Zerowalker** [reply]
I think I got it to work after playing around a bit.
I'm just not too familiar with the setup as it seems to have changed between different Core versions.

But I seem to be on the right track so i will say this is solved:)

Many thanks!
